public/app/plugins/datasource/elasticsearch/components/QueryEditor/BucketAggregationsEditor/SettingsEditor/DateHistogramSettingsEditor.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { ComponentProps, useState } from 'react';
+import { InlineField, Input, Select } from '@grafana/ui';
+import { DateHistogram } from '../aggregations';
+import { bucketAggregationConfig } from '../utils';
+import { useDispatch } from '../../../../hooks/useStatelessReducer';
+import { SelectableValue } from '@grafana/data';
+import { changeBucketAggregationSetting } from '../state/actions';
+import { inlineFieldProps } from '.';
+
+type IntervalOption = SelectableValue<string>;
+
+const defaultIntervalOptions: IntervalOption[] = [
+  { label: 'auto', value: 'auto' },
+  { label: '10s', value: '10s' },
+  { label: '1m', value: '1m' },
+  { label: '5m', value: '5m' },
+  { label: '10m', value: '10m' },
+  { label: '20m', value: '20m' },
+  { label: '1h', value: '1h' },
+  { label: '1d', value: '1d' },
+];
+
+const hasValue = (searchValue: IntervalOption['value']) => ({ value }: IntervalOption) => value === searchValue;
+
+const isValidNewOption: ComponentProps<typeof Select>['isValidNewOption'] = (
+  inputValue,
+  _,
+  options: IntervalOption[]
+) => {
+  // TODO: would be extremely nice here to allow only template variables and values that are
+  // valid date histogram's Interval options
+  const valueExists = options.some(hasValue(inputValue));
+  // we also don't want users to create "empty" values
+  return !valueExists && inputValue.trim().length > 0;
+};
+
+const optionStartsWithValue: ComponentProps<typeof Select>['filterOption'] = (option: IntervalOption, value) =>
+  option.value?.startsWith(value) || false;
+
+interface Props {
+  bucketAgg: DateHistogram;
+}
+
+const getInitialState = (initialValue?: string): IntervalOption[] => {
+  return defaultIntervalOptions.concat(
+    defaultIntervalOptions.some(hasValue(initialValue))
+      ? []
+      : {
+          value: initialValue,
+          label: initialValue,
+        }
+  );
+};
+
+export const DateHistogramSettingsEditor = ({ bucketAgg }: Props) => {
+  const dispatch = useDispatch();
+
+  const [intervalOptions, setIntervalOptions] = useState<IntervalOption[]>(
+    getInitialState(bucketAgg.settings?.interval)
+  );
+
+  const addIntervalOption = (value: string) => setIntervalOptions([...intervalOptions, { value, label: value }]);
+
+  const handleIntervalChange = (v: string) => dispatch(changeBucketAggregationSetting(bucketAgg, 'interval', v));
+
+  return (
+    <>
+      <InlineField label="Interval" {...inlineFieldProps}>
+        <Select<string>
+          inputId="asd"
```

Comment:
That's a bit disappointing. Why not `inputId=lol` or `inputId=yolo`?

Reply:
lol, forgot to remove them ahah (but that's also why it's still in draft :D )